source/modulo_components/modulo_components/component_interface.py

```diff
@@ -250,7 +250,17 @@ def __validate_parameter(self, parameter: sr.Parameter) -> bool:
             if value <= 0 or value > sys.float_info.max:
                 self.get_logger().error("Value for parameter 'rate' has to be a positive finite number.")
                 return False
-        return self.on_validate_parameter_callback(parameter)
+        try:
+            result = self.on_validate_parameter_callback(parameter)
+        except AttributeError:
+            self.get_logger().error("Attribute error during 'on_validate_parameter_callback', declare necessary "
+                                    "attributes before __init__ of the parent class")
```

Comment:
Be careful of "run-on sentences" where a comma is used between two complete sentences. It's very common in casual writing nowadays but it's technically not correct.
```suggestion
            self.get_logger().error("Attribute error during 'on_validate_parameter_callback'. Declare necessary "
                                    "attributes before __init__ of the parent class")
```